website/src/components/inputTable/ColumnsTable.jsx

```diff
@@ -1,65 +1,87 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import HoverVisible from "../hoverVisible/HoverVisible";
-import TextField from "@mui/material/TextField";
-import { Checkbox, MenuItem, Select } from "@mui/material";
+import Checkbox from '@mui/material/Checkbox';
+import InputString from "../inputs/InputString";
+import { isBoolean, isNotNull, isNumber } from "../../utils/validatorUtils";
+import InputDropDown from "../inputs/InputDropDown";
 
 const ColumnsTable = (props) => {
     const [visible, setVisible] = useState(false);
-
+    const [columns, setColumns] = useState([]);
     const items = [];
+    useEffect(() => {
+        if(columns.length !== props.value.length){
+            setColumns(props.value)
+        }
+        
+    }, [props.value.length])
```

Comment:
so we don't with to capture change inside a specific field. only if fields were removed or added?

Reply:
yes, just to update the state that holds these values